rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -148,6 +142,61 @@ for field in spec.fields:
   }
 @[end for]@
 
+@{
+def gen_init_func(funcname, use_defaults, use_zeros):
+    init_strs = []
+    for field in spec.fields:
+        # if default value is available set it in initializer list
+        if field.type.is_primitive_type():
+            # Primitive type.
+            if not field.type.is_array:
+                cpp_value = None
+                if use_defaults and field.default_value is not None:
+                    cpp_value = value_to_cpp(field.type, field.default_value)
+                elif use_zeros:
+                    if field.type.type == 'string':
+                        cpp_value = '""';
+                    else:
+                        cpp_value = '0'
+                if cpp_value is not None:
+                    init_strs.append('this->%s = %s;' % (field.name, cpp_value))
+            # Arrays are represented either by a std::vector or by a BoundedVector.
+            # In both cases they are not primitive types, so their constructors
+            # will automatically be run and we don't have to do anything.
+        else:
+            if not field.type.is_array:
+                init_strs.append('this->%s.%s();' % (field.name, funcname))
+            # Arrays are represented either by a std::vector or by a BoundedVector.
+            # In both cases they are not primitive types, so their constructors
+            # will automatically be run and we don't have to do anything.
+
+    return init_strs
+
+init_all_strings = gen_init_func('initialize_all', True, True)
+init_zero_strings = gen_init_func('zero_initialize', False, True)
+init_default_strings = gen_init_func('initialize_defaults_only', True, False)
+}@
+  void initialize_all()
+  {
+@[for initstr in init_all_strings]@
+    @(initstr)
+@[end for]@
+  }
+
+  void zero_initialize()
```

Comment:
To be consistent should this be named `initialize_zero` instead?

Reply:
Yeah, definitely, will fix.